core/src/main/scala/org/apache/spark/shuffle/ShuffleMergerDiscovery.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import org.apache.spark.annotation.{Experimental, Since}
+import org.apache.spark.storage.BlockManagerId
+
+/**
+ * :: Experimental ::
+ * Pluggable interface to get merger locations for Push based shuffle.
+ */
+@Experimental
+@Since("3.1.0")
+trait ShuffleMergerDiscovery {
```

Comment:
With pluggable API, we should expose configurations to set the implementation for this API.

Reply:
Can we get more feedback from OS side as to how to design this API?